programs/steward/src/state/steward_state.rs

```diff
@@ -383,25 +412,24 @@ impl StewardState {
         num_epochs_between_scoring: u64,
     ) -> Result<()> {
         if current_epoch >= self.next_cycle_epoch {
-            self.state_tag = StewardStateEnum::ComputeScores;
             self.reset_state_for_new_cycle(
                 current_epoch,
                 current_slot,
                 num_epochs_between_scoring,
             )?;
-        } else if current_epoch > self.current_epoch {
+        } else if self.has_flag(RESET_TO_IDLE) {
             self.state_tag = StewardStateEnum::Idle;
             self.progress = BitMask::default();
-            self.rebalance_completed = false.into();
+            // NOTE: RESET_TO_IDLE is cleared in the Idle transition
         } else if self.progress.is_complete(self.num_pool_validators)? {
             self.state_tag = StewardStateEnum::Idle;
-            self.rebalance_completed = true.into();
+            self.set_flag(REBALANCE);
         }
         Ok(())
     }
 
     /// Update internal state when transitioning to a new cycle, and ComputeScores restarts
-    fn reset_state_for_new_cycle(
+    pub fn reset_state_for_new_cycle(
```

Comment:
remove this pub?

Reply:
10-4 that was an artifact from a different solution